Userland/Libraries/LibWeb/Selection/Selection.cpp

```diff
@@ -131,8 +131,8 @@ String Selection::direction() const
 // https://w3c.github.io/selection-api/#dom-selection-getrangeat
 WebIDL::ExceptionOr<JS::GCPtr<DOM::Range>> Selection::get_range_at(unsigned index)
 {
-    // The method must throw an IndexSizeError exception if index is not 0, or if this is empty.
-    if (index != 0 || is_empty())
+    // The method must throw an IndexSizeError exception if index is not 0, or if this is empty or either focus or anchor is not in the document tree.
+    if (index != 0 || is_empty() || (!focus_node()->in_a_document_tree() || !anchor_node()->in_a_document_tree()))
```

Comment:
should we null check focus_node and anchor_node before calling in_a_document_tree()?

Reply:
Sounds correct.
I'm sorry but it's not clear to me what the behaviour should be in the event of a nullptr for focus_node / anchor_node?
Both 
```
!(focus_node() && focus_node()->in_a_document_tree())
focus_node() && !focus_node()->in_a_document_tree()
```
(Former throws exception for null whilst latter should pass) causes the test to fail.

Any idea what would be correct?

Reply:
I was thinking about something like:
```cpp
auto is_focus_in_document_tree = focus_node() && focus_node()->in_a_document_tree();
auto is_anchor_in_document_tree = anchor_node() && anchor_node()->in_a_document_tree();
if (index != 0 || is_empty() || !is_focus_in_document_tree || !is_anchor_in_document_tree) {}
```

Reply:
Thanks for the speedy response. That's what I had been considering albeit more in lined. (Don't know the vibes here).
However the test: Tests/LibWeb/Text/input/set-selection-inside-shadow-root.html
fails. Checked that it passes for me in Firefox. This is my 1st PR so lack the broader context to investigate whether the test is wrong given that it passed previously and seems fine in other browsers.